packages/sit-onyx/src/i18n/index.ts

```diff
@@ -0,0 +1,135 @@
+import type { DeepPartial, ObjectToDottedStrings, TranslationValue } from "@/types";
+import { computed, inject, provide, unref, type InjectionKey, type MaybeRef } from "vue";
+import enUS from "./locales/en-US.json";
+
+/** Available translations that are used by Onyx components. */
+export type Translation = typeof enUS;
+
+export type ProvideI18nOptions = {
+  /**
+   * Current locale / language to use.
+   * If a ref is passed (e.g. the locale from the `vue-i18n` package)
+   * all Onyx messages will be updated if it changes (if locale is supported).
+   * If a message is missing for your currently set locale, English will be used as fallback.
+   *
+   * @default "en-US"
+   */
+  locale?: MaybeRef<string>;
+  /**
+   * Available translations / messages. English is always supported. For build-in translations, see:
+   * https://onyx.schwarz/i18n/
+   *
+   * @example
+   * ```ts
+   * import deDE from "sit-onyx/locales/de-DE.json";
+   * {
+   *   messages: {
+   *     // English is always supported so we don't need to add it here
+   *     'de-DE': deDE
+   *   }
+   * }
+   * ```
+   */
+  messages?: Record<string, DeepPartial<Translation> | Record<string, TranslationValue>>;
+};
+
+const I18N_INJECTION_KEY = Symbol() as InjectionKey<ReturnType<typeof createI18n>>;
+
+/**
+ * Creates a new i18n instance.
+ */
+const createI18n = (options?: ProvideI18nOptions) => {
+  /**
+   * Current locale.
+   * @default "en-US"
+   */
+  const locale = computed(() => unref(options?.locale) ?? "en-US");
+
+  const messages = computed(() => {
+    if (options?.messages && locale.value in options.messages) {
+      return options.messages[locale.value];
+    }
+    return enUS;
+  });
+
+  /**
+   * Gets the translation for the given key.
+   * If message is not found for current locale, English fallback will be used.
+   * @param placeholders Placeholders that will be replaced in the message string
+   * For pluralization, you must provide the placeholder `n`.
+   */
+  const t = computed(() => {
+    return (
+      key: ObjectToDottedStrings<Translation>,
+      placeholders: Record<string, string | number | undefined> = {},
+    ): string => {
+      // use English message as fallback
+      let message = resolveMessage(key, messages.value) ?? resolveMessage(key, enUS) ?? "";
+
+      if (placeholders) {
+        Object.entries(placeholders).forEach(([key, value]) => {
+          if (value === undefined) return;
```

Comment:
didn't you say for undefined placeholder values, a fallback was planned -> to show the key instead? like that it would still show e.g. {value}, like we saw in the screenshot from an older comment in this PR.

Or is this intended in hopes the dev might notice something is wrong?

Reply:
ah, the case in the older comment was that no placeholders were passed, so it's a different issue. 

but I suppose the behavior would be the same here if you pass e.g. undefined for "value"

Reply:
I was referring to https://github.com/SchwarzIT/onyx/pull/75/files/f9ea0c2113dd710d03fbeae1bb3392fc4ea58c99#r1450141113

Reply:
maybe the fallback could be just `""` or just an empty string ¯\_(ツ)_/¯

Reply:
Yes the placeholders are kept as seen here https://github.com/SchwarzIT/onyx/pull/75#discussion_r1450713080

Or did I got your comment wrong?

Reply:
If the translation is `Hello {firstName}` and you don't pass a value for firstName, the resulting string should be
`Hello {firstName}`

Reply:
Finding: `vue-i18n` removes empty placeholders so we should align and do the same